### PR TITLE
simplify common:

### PR DESCRIPTION
**Pull request by bertsky**
- move resolving @imageFilename into `image_from_page`, too
- move OcrdExif there as well (cannot be done outside
  anymore, because AlternativeImage is not as reliable)
- unify image_from_* for all segments below page into
  `image_from_segment`, check whether rotation/orientation
  is present in segment/parent dynamically

### CODE REVIEW

**Comment by bertsky** [comment]
As discussed earlier, this is another proposal for a somewhat simpler API with less code duplication in the implementation.

**Comment by bertsky** [comment]
@kba I am agnostic over which variant makes the race for core.

File: ocrd_tesserocr/binarize.py [hunk 1]
```diff
@@ -23,8 +23,7 @@
 from .config import TESSDATA_PREFIX, OCRD_TOOL
 from .common import (
     image_from_page,
-    image_from_region,
-    image_from_line,
+    image_from_segment,
```
Comment (wrznr) [inline ocrd_tesserocr/binarize.py]: I like it!

File: ocrd_tesserocr/binarize.py [hunk 2]
```diff
@@ -72,16 +71,15 @@ def process(self):
                                                                          value=self.parameter[name])
                                                                for name in self.parameter.keys()])]))
                 page = pcgts.get_Page()
-                page_image = self.workspace.resolve_image_as_pil(page.imageFilename)
+                page_image, page_xywh, _ = image_from_page(
```
Comment (wrznr) [inline ocrd_tesserocr/binarize.py]: That is another function which has to be moved to `core`, right?

Comment (wrznr) [reply ocrd_tesserocr/binarize.py]: Exactly. But a function from the workspace module was used prior to this PR. Treat it as a reminder. No worries.

Comment (bertsky) [reply ocrd_tesserocr/binarize.py]: Indeed, we would want adopters to _forget_ all about `workspace.resolve_image_as_pil` and use the 2 new functions instead. Maybe we can even _hide_ the less abstract functions completely?

Comment (kba) [reply ocrd_tesserocr/binarize.py]: `workspace.resolve_image_as_pil` could be marked as deprecated and phased out in the coming releases.

> Maybe we can even hide the less abstract functions completely?

What do you mean with "hide"?

Comment (bertsky) [reply ocrd_tesserocr/binarize.py]: Sounds good. Hide: that would have been renaming with leading underscore, but of course, we do not want existing implementations to break.

**Comment by bertsky** [comment]
I will merge now and update the ocropy side accordingly.

**Pull request merged**